Minor updates

lkorbel opened the pull request:
1. New type added to MConfig: QUrl
1. Added minimum CMake version (details in commit message)

Commit:
Add QUrl type to MConfig

lkorbel commented:
@sierdzio sending one small addition that should be useful (QUrl support), change number 2 is not relevant anymore

sierdzio commented:
> @sierdzio sending one small addition that should be useful (QUrl support), change number 2 is not relevant anymore

Thanks, I'll check it out when I find some time.

I was sure I merged this before, hmmm. Must have slipped under the radar somehow, or repos got desynchronised :|

Commit:
Fix bug in mconfig-crypto option

Previously, if mcrypto was added to the project, mconfig will use it regardless
of related cmake option value which may lead to runtime crashes (and did in my case).

Commit:
Port code to Qt6

There are few depracated methods in QVariant. Code remain back compatible with Qt5.
Added CMakeLists for Qt6.
